include/deal.II/lac/constraint_matrix.templates.h

```diff
@@ -855,16 +856,7 @@ ConstraintMatrix::distribute (VectorType &vec) const
   // the last else is for the simple case (sequential vector)
   const IndexSet vec_owned_elements = vec.locally_owned_elements();
 
-  // FIXME: This has to be refactored into a typetrait
-  if ((typeid(vec) != typeid(Vector<double>)) &&
-      (typeid(vec) != typeid(Vector<float>)) &&
-      (typeid(vec) != typeid(Vector<std::complex<double> >)) &&
-      (typeid(vec) != typeid(BlockVector<double>)) &&
-      (typeid(vec) != typeid(BlockVector<float>)) &&
-      (typeid(vec) != typeid(BlockVector<std::complex<double> >)) &&
-      (typeid(vec) != typeid(LinearAlgebra::Vector<double>)) &&
-      (typeid(vec) != typeid(LinearAlgebra::Vector<float>)) &&
-      (typeid(vec) != typeid(LinearAlgebra::Vector<std::complex<double> >)))
+  if ( !dealii::is_serial_vector< VectorType >::value )
```

Comment:
make it more obvious by saying `...::value == false` -- by the time I've read all the way down to the `value`, I've forgotten already that the class name contains a predicate (`is_..`).